RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -102,36 +101,18 @@ - (void)setAnnotations:(NSMutableDictionary *)annotations
 - (void)updateAnnotations
 {
     if (_newAnnotations) {
-        NSArray *oldKeys = [_annotations allKeys];
-        NSArray *newKeys = [_newAnnotations allKeys];
-
         // Take into account any already placed pins
-        if (oldKeys.count) {
+        if (_annotations.count) {
             NSMutableArray *removeableKeys = [NSMutableArray array];
-            for (NSValue *oldKey in oldKeys){
-                // Collect all keys that existed in "oldKeys" but not in "newKeys"
-                // anymore, so they can be removed
-                if (![newKeys containsObject:oldKey]){
-                    [removeableKeys addObject:oldKey];
-                }
-            }
-
-            // Remove each of the "removableKeys" from the map
-            if (removeableKeys.count){
-                NSArray *removed = [_annotations objectsForKeys:removeableKeys notFoundMarker:[NSNull null]];
-                [_map removeAnnotations: removed];
+            for (RCTMGLAnnotation *oldKey in _annotations){
```

Comment:
Instead of this `for` loop, call `[_map removeAnnotations:_annotations]`, then call `[_annotations removeAllObjects]`.